ddtrace/tracer/span.go

```diff
@@ -283,10 +304,14 @@ func (s *span) setMetric(key string, v float64) {
 	}
 	delete(s.Meta, key)
 	switch key {
+	case ext.ManualKeep:
+		if v == float64(samplerAppSec) {
+			s.setSamplingPriorityLocked(ext.PriorityUserKeep, samplerAppSec, 0)
+		}
 	case ext.SamplingPriority:
```

Comment:
```suggestion
	case ext.SamplingPriority, keySamplingPriority:
```
If you make this change, you should be able to delete that code where you left the `TODO`

Reply:
Hmm, I think it will cause an infinite loop. Because we have `s.setMetric(keySamplingPriority, float64(priority))` in `setSamplingPriorityLocked`.

Reply:
I mean, we can avoid that by removing `s.setMetric(keySamplingPriority, float64(priority))`, and just use `Metric[keySamplingPriority] = p`

Reply:
I'll leave it up to you. But we should either get rid of the `TODO`, or apply the `TODO` and remove that logic there.

Reply:
I removed `TODO` for now. I might apply the `TODO` comment in a separate PR.